### ISSUE TITLE
[UX] Decide whether hiding the primary/secondary tabs when there's only one is a good or a bad thing.

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#6579]
The inconsistency around this has been annoying me for ever! I've been bumping into this UX WFT from time to time, and again now while working in #1456. Here are screenshots from all the "Add xyz" pages under "Structure", where some of the "Add xyz" pages are showing the horizontal tabs, while others don't. These include the page title changes from that other issue, since otherwise it would be to confusing and difficult to discern that these are in fact "Add xyz" pages, but the logic around the primary/secondary has not changed:
<img width="333" alt="image" src="https://github.com/backdrop/backdrop-issues/assets/2423362/dae438af-17b8-44cb-949c-17f9339245f3">
<img width="329" alt="image" src="https://github.com/backdrop/backdrop-issues/assets/2423362/ad6d5220-b1ee-4e00-9398-b432da9130fa">
<img width="321" alt="image" src="https://github.com/backdrop/backdrop-issues/assets/2423362/882ec374-f12a-43ce-81af-132d67045619">
<img width="617" alt="image" src="https://github.com/backdrop/backdrop-issues/assets/2423362/092ef495-5a61-4915-8334-f6c9802d632b">
<img width="537" alt="image" src="https://github.com/backdrop/backdrop-issues/assets/2423362/2a6df182-eba1-4385-8b43-f27a440e0f7c">
<img width="320" alt="image" src="https://github.com/backdrop/backdrop-issues/assets/2423362/50bc0e48-a41c-4678-b519-c2445cd3af3d">
<img width="392" alt="image" src="https://github.com/backdrop/backdrop-issues/assets/2423362/f9eaf635-1ec5-4593-a966-e8f0884020ea">
<img width="311" alt="image" src="https://github.com/backdrop/backdrop-issues/assets/2423362/b050ddf8-420a-4e1a-b30d-29fb1144fa70">
<img width="600" alt="image" src="https://github.com/backdrop/backdrop-issues/assets/2423362/bb42782e-e899-417b-babe-2bba649f10e6">
<img width="274" alt="image" src="https://github.com/backdrop/backdrop-issues/assets/2423362/b657ab89-e5d0-4365-ba12-596cd533866d">

Each time I saw that inconsistency, I was under the impression that it was a bug. It turns out that this is by design, and it happens for both primary and secondary horizontal tabs, depending on the number of items they have:
```php
/**
 * Returns the rendered local tasks at the top level.
 */
function menu_primary_local_tasks() {
  $links = menu_local_tasks(0);
  // Do not display single tabs.
  return ($links['tabs']['count'] > 1 ? $links['tabs']['output'] : '');
}

/**
 * Returns the rendered local tasks at the second level.
 */
function menu_secondary_local_tasks() {
  $links = menu_local_tasks(1);
  // Do not display single tabs.
  return ($links['tabs']['count'] > 1 ? $links['tabs']['output'] : '');
}
```

Should we be always shoeing the horizontal tabs, even if only one of them is available? Should it be done conditionally? (for instance only for the various pages under the "Structure" menu) ...or should we consider making that configurable and provide an option for this in a new "User interface" form under `/admin/config/administration/ui`? (grouped with the settings for the admin bar)

**Comment by NormPlum** [comment]
I don't think this has to do with the number of tabs, but whether any tabs are present on the parent page. For example, here are screenshots of the parent pages:

<img width="414" height="240" alt="Image" src="https://github.com/user-attachments/assets/e7904333-a995-4352-aaee-91ab24d6f432" />
<br />
<img width="359" height="256" alt="Image" src="https://github.com/user-attachments/assets/8c03a56d-a0f1-4c18-a010-4f031a670383" />
<br />
<img width="423" height="318" alt="Image" src="https://github.com/user-attachments/assets/d87da228-8988-434a-908d-a4e2cb2172bc" />
<br />
<img width="706" height="287" alt="Image" src="https://github.com/user-attachments/assets/a0bd9635-8232-4f33-948a-ac72aaaccd24" />


Where your screenshots show no tabs, the parent page has no tabs. Where yours shows tabs, the parent page has tabs.
